agent/utils/utils.go

```diff
@@ -109,11 +110,28 @@ var _time ttime.Time = &ttime.DefaultTime{}
 // If the error is Retriable then that will be used to determine if it should be
 // retried
 func RetryWithBackoff(backoff Backoff, fn func() error) error {
+	return RetryWithBackoffCtx(context.Background(), backoff, fn)
+}
+
+// RetryWithBackoffCtx takes a context, a Backoff, and a function to call that returns an error
+// If the context is done, nil will be returned
+// If the error is nil then the function will no longer be called
+// If the error is Retriable then that will be used to determine if it should be
+// retried
+func RetryWithBackoffCtx(ctx context.Context, backoff Backoff, fn func() error) error {
 	var err error
-	for err = fn(); true; err = fn() {
-		retriable, isRetriable := err.(Retriable)
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		default:
```

Comment:
Neat!! 

Why keep the default block empty? Would it make sense to put `err = fn()`.. etc there?

Reply:
Seemed clearer to me to have it out.  This way it reads that we early return if the context is done, but otherwise keep going.